### PR TITLE
Integrate rules page

### PR DESCRIPTION
**Pull request by sans-harness**
- Adds the `repo-settings-rules-page` to gitness. 
- e2e vid:

https://github.com/user-attachments/assets/99761f23-f9c5-4454-94fa-0d4e69b19c98

### CODE REVIEW

File: packages/playground/src/components/repo-settings/repo-settings-general/repo-settings-general-security.tsx [hunk 1]
```diff
@@ -1,77 +1,102 @@
...
-                {errors.secretScanning.message?.toString()}
-              </FormFieldSet.Message>
-            )}
-          </FormFieldSet.ControlGroup>
+  })
+
+  const onCheckboxChange = (checked: boolean) => {
+    setValue('secretScanning', checked)
+    handleSubmit(data => {
+      console.log(data)
```
Comment (cjlee01) [inline packages/playground/src/components/repo-settings/repo-settings-general/repo-settings-general-security.tsx:43]: remove console.log

File: packages/playground/src/components/repo-settings/repo-settings-general/repo-settings-general-form.tsx [hunk 2]
```diff
@@ -12,102 +12,121 @@ import {
...
-  const handleSelectChange = (fieldName: keyof FormFields, value: string) => {
+  useEffect(() => {
+    if (isSubmitted && isRepoUpdateSuccess) {
+      setTimeout(() => {
+        setIsSubmitted(false)
+      }, 1000)
+    }
+  }, [isSubmitted, isRepoUpdateSuccess])
+
+  console.log(isRepoUpdateSuccess)
```
Comment (cjlee01) [inline packages/playground/src/components/repo-settings/repo-settings-general/repo-settings-general-form.tsx:82]: remove console.log

File: packages/playground/src/components/repo-settings/repo-settings-general/repo-settings-general-form.tsx [hunk 3]
```diff
@@ -12,102 +12,121 @@ import {
...
   }
 
-  const handleAccessChange = (value: '1' | '2') => {
+  const handleAccessChange = (value: AccessLevel) => {
     setValue('access', value, { shouldValidate: true })
   }
-  const onSubmit: SubmitHandler<FormFields> = data => {
+  const onSubmit: SubmitHandler<RepoUpdateFormFields> = data => {
     setIsSubmitted(true)
     console.log(data)
```
Comment (cjlee01) [inline packages/playground/src/components/repo-settings/repo-settings-general/repo-settings-general-form.tsx:93]: remove console.log

File: packages/playground/src/components/repo-settings/repo-settings-general/repo-settings-general-form.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (sans-harness) [reply packages/playground/src/components/repo-settings/repo-settings-general/repo-settings-general-form.tsx]: This is not a part of the callback fn. This is essentially to show the user that the changes have been saved using a styled success button.

**Review by cjlee01 — APPROVED** [review]

**Pull request merged**